src/base/resource_client.js

```diff
@@ -30,7 +30,7 @@ class ResourceClient extends ApiClient {
         const requestOpts = {
             url: this._url(),
             method: 'GET',
-            params: this._params(options),
+            params: options,
```

Comment:
We should make a copy of the options where we're passing the options object directly and not recreating or destructuring. This is asking for weird bugs in user code.